File: app/src/main/java/com/example/donappt5/helpclasses/MyGlobals.java
package com.example.donappt5.helpclasses;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.util.Log;
import android.view.MenuItem;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import com.example.donappt5.BrowseActivity;
import com.example.donappt5.CharitiesMapActivity;
import com.example.donappt5.CharityCreationActivity;
import com.example.donappt5.CharityListActivity;
import com.example.donappt5.ProfileActivity;
import com.example.donappt5.R;
import com.example.donappt5.SettingsActivity;
import com.facebook.AccessToken;
import com.facebook.GraphRequest;
import com.facebook.GraphResponse;
import com.facebook.HttpMethod;
import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.android.material.navigation.NavigationView;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;
import com.google.firebase.firestore.DocumentReference;
import com.google.firebase.firestore.DocumentSnapshot;
import com.google.firebase.firestore.FirebaseFirestore;
import com.squareup.picasso.Picasso;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.List;

import androidx.annotation.NonNull;
import androidx.appcompat.app.ActionBarDrawerToggle;
import androidx.drawerlayout.widget.DrawerLayout;

public class MyGlobals{
    Context mContext;

    // constructor
    public MyGlobals(Context context){
        this.mContext = context;
    }

    public String getUserName(){
        return "test";
    }
    String photourlfromstore;
    public DrawerLayout drawerlayout;

    public boolean isNetworkConnected() {
        ConnectivityManager cm = (ConnectivityManager) mContext.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo ni = cm.getActiveNetworkInfo();
        if (ni == null) {
            // There are no active networks.
            return false;
        } else
            return true;
    }

    public void setupNavDrawer(Context ctx, Activity activity, DrawerLayout gdrl) {
        drawerlayout = gdrl;
        ActionBarDrawerToggle actionbartoggle = new ActionBarDrawerToggle(activity, drawerlayout, R.string.Open, R.string.Close);

        drawerlayout.addDrawerListener(actionbartoggle);
        actionbartoggle.syncState();


        NavigationView navigationview = (NavigationView) activity.findViewById(R.id.nv);
        navigationview.setNavigationItemSelectedListener(new NavigationView.OnNavigationItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(@NonNull MenuItem item) {
                int id = item.getItemId();
                switch (id) {
                    case R.id.account:
                        Toast.makeText(ctx, "My Account", Toast.LENGTH_SHORT).show();
                        Intent intent = new Intent(ctx, ProfileActivity.class);
                        activity.startActivity(intent);
                        break;
                    case R.id.settings:
                        Toast.makeText(ctx, "Settings", Toast.LENGTH_SHORT).show();
                        Intent intent2 = new Intent(ctx, SettingsActivity.class);
                        activity.startActivity(intent2);
                        break;
                    case R.id.create:
                        Toast.makeText(ctx, "Create Charity", Toast.LENGTH_SHORT).show();
                        Intent intent1 = new Intent(ctx, CharityCreationActivity.class);
                        activity.startActivity(intent1);
                        break;
                    case R.id.map:
                        Toast.makeText(ctx, "Charities map", Toast.LENGTH_SHORT).show();
                        Intent intent3 = new Intent(ctx, CharitiesMapActivity.class);
                        activity.startActivity(intent3);
                        break;
                    case R.id.browse:
                        Toast.makeText(ctx, "Browse", Toast.LENGTH_SHORT).show();
                        Intent intent4 = new Intent(ctx, BrowseActivity.class);
                        activity.startActivity(intent4);
                        break;
                    case R.id.list:
                        Toast.makeText(ctx, "Charity List", Toast.LENGTH_SHORT).show();
                        Intent intent5 = new Intent(ctx, CharityListActivity.class);
                        activity.startActivity(intent5);
                        break;
                    default:
                        return true;
                }
                return true;

            }
        });

        FirebaseUser user = FirebaseAuth.getInstance().getCurrentUser();

        View header = navigationview.getHeaderView(0);
        final ImageView ivinHeader = header.findViewById(R.id.nav_header_imageView);
        TextView tvinHeader = header.findViewById(R.id.nav_header_textView);

        FirebaseFirestore db = FirebaseFirestore.getInstance();
        final DocumentReference docRef = db.collection("users").document(user.getUid());
        docRef.get().addOnCompleteListener(new OnCompleteListener<DocumentSnapshot>() {
            @Override
            public void onComplete(@NonNull Task<DocumentSnapshot> task) {
                if (task.isSuccessful()) {
                    DocumentSnapshot document = task.getResult();
                    photourlfromstore = document.getString("photourl");
                    Picasso.with(ctx).load(photourlfromstore).fit().into(ivinHeader);
                } else {
                    Log.d("fuck", "get failed with ", task.getException());
                }
            }
        });

        if (user != null) {
            if (photourlfromstore != null) {
                Picasso.with(ctx).load(photourlfromstore).fit().into(ivinHeader);
            } else {
                if (user.getPhotoUrl() != null) {
                    //Picasso.get().load(user.getPhotoUrl()).into(ivinHeader);
                    Picasso.with(ctx).load(user.getPhotoUrl().toString()).fit().into(ivinHeader);
                }
            }
            tvinHeader.setText(user.getDisplayName());
        }
    }

    public List<String> getFriendsList() {
        final List<String> friendslist = new ArrayList<String>();
        new GraphRequest(AccessToken.getCurrentAccessToken(), "/me/friends", null, HttpMethod.GET, new GraphRequest.Callback() {
            public void onCompleted(GraphResponse response) {
                /* handle the result */
                Log.e("Friends List: 1", response.toString());
                try {
                    JSONObject responseObject = response.getJSONObject();
                    JSONArray dataArray = responseObject.getJSONArray("data");

                    for (int i = 0; i < dataArray.length(); i++) {
                        JSONObject dataObject = dataArray.getJSONObject(i);
                        String fbId = dataObject.getString("id");
                        String fbName = dataObject.getString("name");
                        Log.e("FbId", fbId);
                        Log.e("FbName", fbName);
                        friendslist.add(fbId);
                        Log.d("friendslist", "fbid: " + fbId);
                    }
                    Log.e("fbfriendList", friendslist.toString());
                    List<String> list = friendslist;
                    String friends = "";
                    if (list != null && list.size() > 0) {
                        friends = list.toString();
                        if (friends.contains("[")) {
                            friends = (friends.substring(1, friends.length() - 1));
                        }
                    }
                } catch (JSONException e) {
                    e.printStackTrace();
                } finally {
                    Log.d("friendslist", "hideLoadingProgress();");
                }
            }
        }).executeAsync();
        return friendslist;
    }
}

File: app/src/main/java/com/example/donappt5/helpclasses/MyGlobals.kt
package com.example.donappt5.helpclasses

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.net.ConnectivityManager
import android.transition.Fade
import android.util.Log
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.ActionBarDrawerToggle
import androidx.drawerlayout.widget.DrawerLayout
import com.example.donappt5.*
import com.example.donappt5.QRStuff.QRGenerateActivity
import com.facebook.AccessToken
import com.facebook.GraphRequest
import com.facebook.HttpMethod
import com.google.android.material.bottomnavigation.BottomNavigationView
import com.google.android.material.bottomnavigation.LabelVisibilityMode.LABEL_VISIBILITY_UNLABELED
import com.google.android.material.navigation.NavigationBarView
import com.google.android.material.navigation.NavigationView
import com.google.android.material.tabs.TabLayout
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.firestore.FirebaseFirestore
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.activity_charitylist.view.*
import kotlinx.android.synthetic.main.activity_owned_charity_list.*
import okhttp3.internal.threadFactory
import org.json.JSONException

class MyGlobals     // constructor
    (var mContext: Context) {
    val userName: String
        get() = "test"
    var photourlfromstore: String? = null
    var drawerlayout: DrawerLayout? = null

    // There are no active networks.
    val isNetworkConnected: Boolean
        get() {
            val cm = mContext.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
            val ni = cm.activeNetworkInfo
            return ni != null
        }

    fun setupBottomNavigation(
        ctx: Context,
        activity: Activity,
        view: BottomNavigationView
    ) {
        view.labelVisibilityMode = NavigationBarView.LABEL_VISIBILITY_UNLABELED;

        activity.overridePendingTransition(0,0)

        setSelectedItem(activity, view)

        view.setOnItemSelectedListener { item ->
            when(item.itemId) {
                R.id.charitylist -> {
                    val intent5 = Intent(ctx, CharityListActivity::class.java)
                    activity.startActivity(intent5)
                    true
                }
                R.id.map -> {
                    val intent3 = Intent(ctx, CharitiesMapActivity::class.java)
                    activity.startActivity(intent3)
                    true
                }
                R.id.donations -> {
                    val intent6 = Intent(ctx, OwnedCharityListActivity::class.java)
                    activity.startActivity(intent6)
                    true
                }
                R.id.profile -> {
                    val intent = Intent(ctx, ProfileActivity::class.java)
                    activity.startActivity(intent)
                    true
                }
                else -> false
            }
        }
    }

    fun setSelectedItem(activity: Activity,
        view: BottomNavigationView) {
        view.menu.findItem(when(activity) {
            is CharityListActivity -> { R.id.charitylist }
            is BrowseActivity -> { R.id.charitylist }
            is CharitiesMapActivity -> { R.id.map }
            is CharityActivity -> { R.id.charitylist }
            is CharityCreationActivity -> { R.id.donations }
            is CharityEditActivity -> { R.id.donations }
            is OwnedCharityListActivity -> { R.id.donations }
            is ProfileActivity -> { R.id.profile }
            is SettingsActivity -> { R.id.profile }
            else -> R.id.charitylist
        }).isChecked = true
    }

    /* handle the result */
    val friendsList: List<String>
        get() {
            val friendslist: MutableList<String> = ArrayList()
            GraphRequest(
                AccessToken.getCurrentAccessToken(), "/me/friends", null, HttpMethod.GET
            ) { response -> /* handle the result */
                Log.e("Friends List: 1", response.toString())
                try {
                    val responseObject = response.jsonObject
                    val dataArray = responseObject.getJSONArray("data")
                    for (i in 0 until dataArray.length()) {
                        val dataObject = dataArray.getJSONObject(i)
                        val fbId = dataObject.getString("id")
                        val fbName = dataObject.getString("name")
                        Log.e("FbId", fbId)
                        Log.e("FbName", fbName)
                        friendslist.add(fbId)
                        Log.d("friendslist", "fbid: $fbId")
                    }
                    Log.e("fbfriendList", friendslist.toString())
                    val list: List<String> = friendslist
                    var friends = ""
                    if (list != null && list.size > 0) {
                        friends = list.toString()
                        if (friends.contains("[")) {
                            friends = friends.substring(1, friends.length - 1)
                        }
                    }
                } catch (e: JSONException) {
                    e.printStackTrace()
                } finally {
                    Log.d("friendslist", "hideLoadingProgress();")
                }
            }.executeAsync()
            return friendslist
        }

}